### PR TITLE
Move response Converter lookup to happen during method parsing.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: retrofit-converters/protobuf/src/test/java/retrofit/ProtoConverterTest.java [hunk 1]
```diff
@@ -81,9 +81,8 @@
     ByteString encoded = ByteString.decodeBase64("Cg4oNTE5KSA4NjctNTMwOQ==");
     server.enqueue(new MockResponse().setBody(new Buffer().write(encoded)));
 
-    Call<?> call = service.wrongClass();
     try {
-      call.execute();
+      service.wrongClass();
```
Comment (JakeWharton) [inline retrofit-converters/protobuf/src/test/java/retrofit/ProtoConverterTest.java]: Assertions failing earlier in API interaction.

File: retrofit/src/main/java/retrofit/OkHttpCall.java [hunk 2]
```diff
@@ -141,36 +140,15 @@ private void callSuccess(Response<T> response) {
...
-        //noinspection unchecked
-        return Response.success((T) bufferedBody, rawResponse);
-      } finally {
-        closeQueitly(rawBody);
-      }
-    }
-
     ExceptionCatchingRequestBody catchingBody = new ExceptionCatchingRequestBody(rawBody);
     try {
-      //noinspection unchecked
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/OkHttpCall.java]: Friendly generics!

File: retrofit/src/main/java/retrofit/MethodInfo.java [hunk 3]
```diff
@@ -85,8 +86,8 @@
     this.method = method;
     this.adapterFactory = adapterFactory;
     this.converterFactory = converterFactory;
-    parseResponseType();
     parseMethodAnnotations();
+    parseResponseType();
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/MethodInfo.java]: This gross parsing order dependency on this highly-mutable object will die soon.

File: retrofit/src/test/java/retrofit/ToStringConverterFactory.java [hunk 4]
```diff
@@ -21,23 +21,20 @@
...
-  static class StringConverter implements Converter<String> {
+  static class StringConverter implements Converter<Object> {
     @Override public String fromBody(ResponseBody body) throws IOException {
       return body.string();
     }
 
-    @Override public RequestBody toBody(String value) {
-      return RequestBody.create(MEDIA_TYPE, value);
+    @Override public RequestBody toBody(Object value) {
+      return RequestBody.create(MEDIA_TYPE, String.valueOf(value));
```
Comment (swankjesse) [inline retrofit/src/test/java/retrofit/ToStringConverterFactory.java]: "null" . . . probably wants a test

Comment (JakeWharton) [inline retrofit/src/test/java/retrofit/ToStringConverterFactory.java]: null will never makes it to a converter

**Pull request merged**